app/src/main/java/com/yacgroup/yacguide/database/RockDao.kt

```diff
@@ -17,31 +17,43 @@
 
 package com.yacgroup.yacguide.database
 
-import android.arch.persistence.room.Dao
-import android.arch.persistence.room.Insert
-import android.arch.persistence.room.OnConflictStrategy
-import android.arch.persistence.room.Query
+import android.arch.persistence.room.*
 
 @Dao
 interface RockDao {
-    @get:Query("SELECT * FROM Rock")
+    @get:Query(Rock.SELECT_ALL)
     val all: List<Rock>
 
-    @Query("SELECT * FROM Rock WHERE parentId = :parentId ORDER BY nr")
+    @Query("${Rock.SELECT_ALL} WHERE parentId = :parentId ORDER BY nr")
     fun getAll(parentId: Int): List<Rock>
 
-    @Query("SELECT * FROM Rock WHERE id = :id")
+    @Query("${Rock.SELECT_ALL} ${Sector.JOIN_ON} Rock.parentId ${Sector.FOR_REGION} :regionId")
+    fun getAllInRegion(regionId: Int): List<Rock>
+
+    @Query("${Rock.SELECT_ALL} ${Sector.JOIN_ON} Rock.parentId ${Region.JOIN_ON} Sector.parentId ${Region.FOR_COUNTRY} :countryName")
+    fun getAllInCountry(countryName: String): List<Rock>
+
+    @Query("${Rock.SELECT_ALL} WHERE id = :id")
     fun getRock(id: Int): Rock?
 
-    @Query("SELECT Rock.* FROM Rock JOIN Route ON Rock.id = Route.parentId JOIN Ascend ON Route.id = Ascend.routeId WHERE Ascend.id = :ascendId")
+    @Query("${Rock.SELECT_ALL} JOIN Route ON Rock.id = Route.parentId JOIN Ascend ON Route.id = Ascend.routeId WHERE Ascend.id = :ascendId")
```

Comment:
Why do you not use the `JOIN_ON` constants here?

Reply:
Because the Route.JOIN_ON refers to joining upwards on Route.Id. Here we join downwards on Route.parentId